lib/msal-browser/src/controllers/NestedAppAuthController.ts

```diff
@@ -0,0 +1,435 @@
+        const currentAccount = this.operatingContext.getActiveAccount();
+        if (currentAccount !== undefined) {
+            if (currentAccount.username === userName) {
+                return this.nestedAppAuthAdapter.fromNaaAccountInfo(
+                    currentAccount
+                );
+            } else {
+                return null;
+            }
+        } else {
+            return null;
+        }
+    }
+    getAllAccounts(): AccountInfo[] {
+        throw new Error("Method not implemented.");
+    }
+    handleRedirectPromise(
+        hash?: string | undefined // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<AuthenticationResult | null> {
+        throw new Error("Method not implemented.");
+    }
+    loginPopup(
+        request?: PopupRequest | undefined // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<AuthenticationResult> {
+        if (request !== undefined) {
+            return this.acquireTokenInteractive(request);
+        } else {
+            throw NestedAppAuthError.createUnsupportedError();
+        }
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    loginRedirect(request?: RedirectRequest | undefined): Promise<void> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    logout(logoutRequest?: EndSessionRequest | undefined): Promise<void> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    logoutRedirect(
+        logoutRequest?: EndSessionRequest | undefined // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<void> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    logoutPopup(
+        logoutRequest?: EndSessionPopupRequest | undefined // eslint-disable-line @typescript-eslint/no-unused-vars
+    ): Promise<void> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    ssoSilent(
+        // eslint-disable-next-line @typescript-eslint/no-unused-vars
+        request: Partial<
+            Omit<
+                CommonAuthorizationUrlRequest,
+                | "requestedClaimsHash"
+                | "responseMode"
+                | "codeChallenge"
+                | "codeChallengeMethod"
+                | "nativeBroker"
+            >
+        >
+    ): Promise<AuthenticationResult> {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    getTokenCache(): ITokenCache {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    getLogger(): Logger {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    setLogger(logger: Logger): void {
+        throw NestedAppAuthError.createUnsupportedError();
+    }
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    setActiveAccount(account: AccountInfo | null): void {
```

Comment:
Something I am thinking about - if this function is called by the Hub in some cases, and by the app developer when running as a standalone page, could it make sense to have this function do nothing and log a warning, rather than `throw` in the nested case? Thinking out loud, a developer would have to fork their code (if/else) to avoid calling this if they detect they are embedded in a rich client.

Reply:
logged a warning and did nothing.